### ISSUE TITLE
OCI runtime exec error: flag provided but not defined: -console-socket

### ISSUE THREAD

**Issue by ianmiell** [google/gvisor#37]
```
gvisor1:2ehal1nT:G7HJX100# docker exec -ti 123 bash
OCI runtime exec failed: /var/lib/docker/runtimes/runsc did not terminate sucessfully: flag provided but not defined: -console-socket
                                                                                                                                     exec [command options] <container-id> <command> [comman
d options] || --process process.json <container-id>
                       
                       
                                                   Where "<container-id>" is the name for the instance of the container and
                                                                                                                           "<command>" is the command to be executed in the container.
                                                                                                                                                                                      "<comm
and>" can't be empty unless a "-process" flag provided.
```

```
gvisor1:2ehal1nT:G7HJX100# docker version
Client:
 Version:      18.03.1-ce
 API version:  1.37
 Go version:   go1.9.5
 Git commit:   9ee9f40
 Built:        Thu Apr 26 07:17:20 2018
 OS/Arch:      linux/amd64
 Experimental: false
 Orchestrator: swarm

Server:
 Engine:
  Version:      18.03.1-ce
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.9.5
  Git commit:   9ee9f40
  Built:        Thu Apr 26 07:15:30 2018
  OS/Arch:      linux/amd64
  Experimental: false
```

```
gvisor1:2ehal1nT:G7HJX100# uname -a
Linux gvisor1 4.4.0-122-generic #146-Ubuntu SMP Mon Apr 23 15:34:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

Code to repro:

https://github.com/ianmiell/shutit-gvisor/blob/master/build.py

**Comment by Ace-Tang** [comment]
I got the same error message
```
$ docker version
Client:
 Version:      18.03.1-ce
 API version:  1.37
 Go version:   go1.9.5
 Git commit:   9ee9f40
 Built:        Thu Apr 26 07:17:20 2018
 OS/Arch:      linux/amd64
 Experimental: false
 Orchestrator: swarm

Server:
 Engine:
  Version:      18.03.1-ce
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.9.5
  Git commit:   9ee9f40
  Built:        Thu Apr 26 07:15:30 2018
  OS/Arch:      linux/amd64
  Experimental: false
```

**Comment by Ace-Tang** [comment]
thanks for your reply, @fvoznika

**Comment by fvoznika** [comment]
It does support `docker exec`, but it doesn't support it when `-ti` option is used.

**Comment by fvoznika** [comment]
Fixed by 106de2182d34197d76fb68863cd4a102ebac2dbb
